javaparser-core/src/main/javacc/java.jj

```diff
@@ -3278,13 +3278,15 @@ Expression EqualityExpression():
  */
 PatternExpr PatternExpression():
 {
+    ModifierHolder modifier;
     ReferenceType type;
     SimpleName name;
 }
 {
-    type = AnnotatedReferenceType()
+    modifier = Modifiers()
+    type = ReferenceType(modifier.annotations)
```

Comment:
Why did you change the type?

Reply:
Looks like a good job to me. Maybe @MysterAitch will take a look at this PR.

Reply:
At first, because `InstanceOfExprTest.annotationsOnTheType_patternExpression` failed otherwise. From my understanding AnnotatedReferenceType is just a helper to parse annotation and then create a ReferenceType with those annotations.
```
ReferenceType AnnotatedReferenceType():
{
    NodeList<AnnotationExpr> annotations;
    ReferenceType type;
}
{
    annotations = Annotations()
    type = ReferenceType(annotations)
    { return type; }
}
```
But since `Modifiers()` also parses annotations the `Annotations()` in the later `AnnotatedReferenceType()` couldn't find anything because annotations where already parsed by `Modifiers()`. That's why I took the annotations from the `ModifierHolder` instead.

Reply:
For me it's good. Good work. I'll give @MysterAitch a chance to look at your proposal as well.